### ISSUE TITLE
E1029 - False positive with multiline YAML string

### ISSUE THREAD

**Issue by suroh1994** [aws-cloudformation/cfn-python-lint#1662]
*cfn-lint version: (`cfn-lint --version`)*
0.35.0

*Description of issue.*
Using ${iot:Connection.Thing.ThingName} inside an AWS::IoT::Policy PolicyDocument is flagged as an embedded parameter outside of Fn::Sub.

Freshly installed cfn-lint this morning. I'm using Python 2.7 and VS Code

* Sample Template:
```
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  TestPolicy:
    Type: AWS::IoT::Policy
    Properties:
      PolicyDocument: >
        {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": "iot:Connect",
              "Resource": "arn:aws:iot:eu-central-1:123456789012:client/${iot:Connection.Thing.ThingName}"
            }
          ]
        }
      PolicyName: ConnectWithThingName
```

In the second example in their developer guide on AWS IoT they use the exact same policy document. https://docs.aws.amazon.com/iot/latest/developerguide/connect-policy.html

* Feature request:
Like with IAM policies, IoT policies shouldn't have variables flagged as embedded parameters.

**Comment by suroh1994** [comment]
I ended up fixing the issue by using Fn::Sub, since I didn't like the hardcoded `region` and `account ID`:

```
Resources:
  TestPolicy:
    Type: AWS::IoT::Policy
    Properties:
      PolicyDocument:
        !Sub >
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": "iot:Connect",
                  "Resource": "arn:aws:iot:${AWS::Region}:${AWS::AccountId}:client/${!iot:Connection.Thing.ThingName}"
                }
              ]
            }
      PolicyName: ConnectWithThingName
```
With the exclamation mark inside ${}, `iot:Connection.Thing.ThingName` will not be replaced.

**Comment by Tro95** [comment]
The problem isn't caused by using a multiline YAML string, the problem is due to the embedded `PolicyDocument` not being parsed into individual paths to the substitution variables. The current logic will ignore E1029 for anything under a `Resource` property, and normally an IAM/IoT policy would be parsed into a path such as `['PolicyDocument', 'Statement', 'Resource', '${iot:Connection.Thing.Thing}']`. Here, the path will equate to `['PolicyDocument', '${iot:Connection.Thing.Thing}']`, as the embedded JSON multiline string will not get parsed in the same way along with the document.

I am hoping this only applies to IAM and IoT `PolicyDocument`s, and will be fixed by #1788.
